rtic-sync/src/channel.rs

```diff
@@ -273,16 +303,21 @@ impl<T, const N: usize> Sender<'_, T, N> {
     fn send_footer(&mut self, idx: u8, val: T) {
         // Write the value to the slots, note; this memcpy is not under a critical section.
         unsafe {
-            ptr::write(
-                self.0.slots.get_unchecked(idx as usize).get() as *mut T,
-                val,
-            )
+            let first_element = self.0.slots.get_unchecked(idx as usize).get_mut();
+            let ptr = first_element.deref().as_mut_ptr();
```

Comment:
What happened in this block?

Reply:
the extra `get_mut()`, `deref()` and `as_ptr()` (all required to fully get the same thing we had before) would've expanded the single line to more than one anyways. I figured not cramming them into the `ptr::read` was more legible, but am open to better suggestions.

Reply:
:+1: